### DISCUSSION TITLE
How do I get to child value

### DISCUSSION THREAD

**Discussion by choonhongyeoh0241** [proyecto26/RestClient#222] [Q&A]
Given the structure of this:

International
   |_  John
          |_ age: 20
          |_name: John
          |_position: school 
   |_ Doe
         |_age: 16
         |_name: Doe
         |_position: store

I am developing 2D game using Unity engine, and I saved my data into firebase using this structure. My question is how do I access to Doe in this example, to delete the whole node of Doe given above. 

![image](https://user-images.githubusercontent.com/103093747/183085204-f1e16657-9e92-49a3-b5e5-5514fbf9d120.png)

**Comment by maifeeulasad** [comment]
First of all, I'm a little confused about your DS, as generally, these types of data are stored in an array. Like: 
```
[
    {
        "name": "name1-john"
    },
    {
        "name": "name2-doe"
    }
]
```
And for this kind of data, the approach is pretty simple:
```
...
[Serializable]
class User
{
    public string name;
}
....
RestClient.GetArray<User>(url).Then(res => {
            foreach (User user in res)
            {
                Debug.Log("user");
                Debug.Log(user.name);
            }
        });
```
But you are using some `Map/Dictionary` type approach. But not sure what is the perfect fit for this. Maybe we can try something like this:
```
...
[Serializable]
class User
{
    public string name;
}

[Serializable]
class UserMap
{
    public Dictionary<string, User> international;
}
...
RestClient.Get<UserMap>(url).Then(res =>
        {
            foreach (string key in res.international.Keys)
            {
                Debug.Log("key");
                Debug.Log(key);
            }
        })
```
I'm not sure if this type of DS has been supported till now or not. But if there is support, then this should be the way, at least I think so.

But it's an excellent question, and I will put some time into it.

Thanks
Maifee Ul Asad

**Reply by choonhongyeoh0241** [reply]
I'm not sure why, but I don't think this is working since I can't debug them. I believe it should be the same logic as Firebase.Database.getReference(), (If I remember correctly)
